src/integrations/jenkins_integration.rs

```diff
@@ -0,0 +1,180 @@
+        let mut auth_headers = Headers::new();
+        auth_headers.set(Authorization(get_basic_credentials(
+            self.username.as_str(),
+            Some(self.password.clone()),
+        )));
+
+        let all_jobs_response: Result<(JenkinsJobResponse, Headers), Error> =
+            get_url_response(&url_string, auth_headers.clone());
+
+        match all_jobs_response {
+            Ok((result, _)) => {
+                let results = result
+                    .jobs
+                    .iter()
+                    .filter(|job| {
+                        job.color != JenkinsJobColor::Disabled
+                            && job.color != JenkinsJobColor::DisabledAnime
+                    })
+                    .map(|job| {
+                        let job_url_string = format!(
+                            "{base}/job/{job}/lastBuild/api/json",
+                            base = self.base_url,
+                            job = job.name
+                        );
+                        let job_response: Result<
+                            (JenkinsBuildResult, Headers),
+                            Error,
+                        > = get_url_response(&job_url_string, auth_headers.clone());
+
+                        match job_response {
+                            Ok((job_result, _)) => {
+                                if job_result.building {
+                                    Ok(JenkinsBuildStatus::Building)
+                                } else {
+                                    let unwrapped_result = job_result.build_result.unwrap();
+                                    Ok(unwrapped_result)
+                                }
+                            }
+                            Err(job_err) => {
+                                warn!("--Jenkins--: HTTP failure when attempting to get job result for job: {}. Error: {}", &job_url_string, job_err);
+                                Err(job_err)
+                            }
+                        }
+                    })
+                    .collect();
+                Ok(results)
+            }
+            Err(err) => Err(err),
+        }
+    }
+}
+
+impl RemoteIntegration for JenkinsIntegration {
+    fn get_red_id(&self) -> u16 {
+        self.r
+    }
+    fn get_green_id(&self) -> u16 {
+        self.g
+    }
+    fn get_blue_id(&self) -> u16 {
+        self.b
+    }
+
+    fn get_status(&mut self) -> RemoteStatus {
+        match self.get_status_internal() {
+            Ok(results) => {
+                let (retrieved, not_retrieved): (
+                    Vec<Result<JenkinsBuildStatus, Error>>,
+                    Vec<Result<JenkinsBuildStatus, Error>>,
+                ) = results.into_iter().partition(|x| x.is_ok());
+
+                let retrieved: Vec<JenkinsBuildStatus> =
+                    retrieved.into_iter().map(|x| x.unwrap()).collect();
+
+                let retrieved_count = retrieved.len();
```

Comment:
This expression has only one usage. Could it be inlined?

Reply:
huh, yeah, it totally could